From c36b0918a8235b6a2d562965e9741760f6f4c1eb Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Mon, 22 Apr 2013 20:49:33 +0200
Subject: [PATCH 54/65] r92su: fix errors found by smatch

main.c:218 r92su_find_wmm_ie() info: loop could be replaced with if statement.
main.c:202 r92su_find_wmm_ie() info: ignoring unreachable code.
main.c:656 r92su_bss_connect_work() warn: variable dereferenced before check 'cfg_bss' (see line 653)
main.c:1106 r92su_bss_build_fw_bss() error: we previously assumed 'tmp' could be null (see line 1106)

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/main.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

--- a/drivers/net/wireless/r92su/main.c
+++ b/drivers/net/wireless/r92su/main.c
@@ -214,8 +214,6 @@ static u8 *r92su_find_wmm_ie(u8 *ies, co
 
 			wmm_ie = wmm_ie + 2 + wmm_ie[1];
 		}
-
-		return NULL;
 	}
 
 	return NULL;
@@ -650,12 +648,12 @@ static void r92su_bss_connect_work(struc
 		goto out;
 
 	cfg_bss = r92su->want_connect_bss;
-	bss_priv = r92su_get_bss_priv(cfg_bss);
 	join_bss = r92su->connect_result;
 
 	if (!cfg_bss || !join_bss)
 		goto out;
 
+	bss_priv = r92su_get_bss_priv(cfg_bss);
 	r92su->connect_result = NULL;
 
 	if (le32_to_cpu(join_bss->bss.ie_length) < 12)
@@ -1103,7 +1101,7 @@ static int r92su_bss_build_fw_bss(struct
 	}
 
 	tmp = r92su_find_ie(ies_data, ies_len, WLAN_EID_DS_PARAMS);
-	if (!tmp && tmp[1] < 1)
+	if (!tmp || tmp[1] < 1)
 		return -EINVAL;
 	fw_bss->config.frequency = cpu_to_le32(tmp[2]);
 
